homeassistant/components/tuya/cover.py

```diff
@@ -238,7 +238,9 @@ def current_cover_position(self) -> int | None:
         if self._current_position is None:
             return None
 
-        if (position := self.device.status.get(self._current_position.dpcode)) is None:
+        if (
+            position := float(self.device.status.get(self._current_position.dpcode))
```

Comment:
This value is a float according to the Tuya specifications, why is a float casting needed here?

Reply:
Although the specification states that it is a float, the library returns a string in some cases. This is the error that is seen in issue #114100 as well. See an example below for a Tuya Batten that's added to the India Data Center:
<img width="1299" alt="image" src="https://github.com/home-assistant/core/assets/1029727/79279c69-3c4b-42b8-a9a5-05ec91d191d7">

Reply:
This should not be fixed in here not in the library, this is an API issue. Have you contact Tuya your device is providing incorrect values on the API?

Reply:
Contacting tuya for anything home-assistant related has been fruitless as the same devices work fine on the smart life app. Based on trying out a variety of white-labeled tuya devices, this issue seems to be common to many devices.

Not to mention that all these devices worked fine prior to the change in how tuya devices were connected to HASS (IoT account vs the current smart-life app pairing).

Tuya devices are fairly common (and is pretty much the de-facto choice) in India, and I've also had to help a couple of friends move certain automations off home assistant since state changes are effectively broken without this fix in place.

My thought is that until Tuya gets around to resolving this issue upstream, there needs to be a way for tuya devices to work with Home Assistant (which this fix achieves). 

A slightly more inconvenient alternative would be to have a fork of the tuya integration that can be installed via HACS. I don't know if it would be straightforward to migrate existing devices to such a fork seamlessly, but that could work as well.

Reply:
> as the same devices work fine on the smart life app. 

That is using a different API.

> Not to mention that all these devices worked fine prior to the change in how tuya devices were connected to HASS (IoT account vs the current smart-life app pairing).

Yup, but that is unfortunately their decision on changing APIs. Please contact Tuya for that, we don't control that.

> My thought is that until Tuya gets around to resolving this issue upstream, there needs to be a way for tuya devices to work with Home Assistant (which this fix achieves).

I don't agree with that. We should fix the root cause. Additionally, this brings in unneeded overhead in type casing in many places. As a matter of fact, you introduced bugs by doing so.